### ISSUE TITLE
GenotypeGVCF AF change in 4.1.7 - 4.1.9

### ISSUE THREAD

**Issue by pbrennan13** [broadinstitute/gatk#6938]
### Instructions

The github issue tracker is for bug reports, feature requests, and API documentation requests. General questions about how to use the GATK, how to interpret the output, etc. should be asked on the [official support forum](http://gatkforums.broadinstitute.org/gatk).
- Search the existing github issues to see if your issue (or something similar) has already been reported.  If the issue already exists, you may comment there to inquire about the progress.
- Determine whether your issue is a **bug report**, a **feature request**, or a **documentation request** (for tool/class javadoc only -- for forum docs please post there)
- Consider if your "issue" is better addressed on the GATK forum: http://gatkforums.broadinstitute.org/gatk  Post there if you have questions about expected tool behavior, output format, unexpected results, or generally any question that does not fit into the categories above
- Use a **concise** yet **descriptive** title
- Choose the corresponding template block below and fill it in, replacing or deleting text in italics (surrounded by `_`) as appropriate
- Delete the other template blocks and this header.

----

### Affected tool(s) or class(es)
GenotypeGVCF

### Affected version(s)
4.1.7
4.1.8
4.1.9

### Description 
Starting with GATK4.1.7, the AF annotation in the changed from '0' to '.'. This change is cause downstream issues with our processing pipeline.

#### Steps to reproduce
CMD using 4.1.6:
gatk GenotypeGVCFs --variant proband_mother_duo.HC.g.vcf.gz -R hs38DH.fa --dbsnp dbsnp151_common.hg38.vcf.gz -O proband_mother_duo_GATK4.1.6.HC.vcf.gz

Looking at one of the sites causing this downstream issue:
CMD
gzcat proband_mother_duo_GATK4.1.6.HC.vcf.gz | grep 83598622
OUTPUT:
chr4 83598622 . AT ATT,A 1337.45 . AC=1,1;AF=0.250,0.250;AN=4;BaseQRankSum=1.26;ClippingRankSum=0.074;DP=145;ExcessHet=4.7712;FS=5.235;GQ_MEAN=625.00;LikelihoodRankSum=1.34;MLEAC=1,1;MLEAF=0.250,0.250;MQ=60.00;MQ0=0;MQRankSum=0.00;NCC=0;NCount=0;QD=10.06;ReadPosRankSum=1.56;SOR=0.375 GT:AD:AF:DP:F1R2:F2R1:GQ:PL 0/2:33,0,35:0.515,0:68:12,15,0:21,18,0:99:713,812,1542,0,731,625 0/1:32,33,0:0.493,0.00:67:9,23,0:19,8,0:99:640,0,588,728,747,1569

CMD using 4.1.7:
gatk GenotypeGVCFs --variant proband_mother_duo.HC.g.vcf.gz -R hs38DH.fa --dbsnp dbsnp151_common.hg38.vcf.gz -O proband_mother_duo_GATK4.1.7.HC.vcf.gz

Looking at one of the sites causing this downstream issue:
CMD
gzcat proband_mother_duo_GATK4.1.7.HC.vcf.gz | grep 83598622
OUTPUT:
chr4 83598622 . AT ATT,A 1337.45 . AC=1,1;AF=0.250,0.250;AN=4;BaseQRankSum=1.26;ClippingRankSum=0.074;DP=145;ExcessHet=4.7712;FS=5.235;GQ_MEAN=625.00;LikelihoodRankSum=1.34;MLEAC=1,1;MLEAF=0.250,0.250;MQ=60.00;MQ0=0;MQRankSum=0.00;NCC=0;NCount=0;QD=10.06;ReadPosRankSum=1.56;SOR=0.375 GT:AD:AF:DP:F1R2:F2R1:GQ:PL 0/2:33,0,35:0.515,.:68:12,15,.:21,18,.:99:713,812,1542,0,731,625 0/1:32,33,0:0.493,0.00:67:9,23,0:19,8,0:99:640,0,588,728,747,1569

The above 4.1.7 site matches 4.1.8 and 4.1.9.

For 4.1.7, The first sample in the VCF lists the 'AF' as "0.515,.", while in version 4.1.6,  AF is represented as "0.515,0"
----

## Feature request

Can the most recent build of 4.1.9 be changed to represent these AF annotations with '0' instead of '.'?

----

**Comment by droazen** [comment]
@ldgauthier Do you know anything about the change described above (from 0 to `.` for the AF annotation)?

**Comment by ldgauthier** [comment]
@pbrennan13 is there a typo in your example output?  The AF in both outputs looks fine to me.

**Comment by pbrennan13** [comment]
I don't think so. The 4.1.6 example shows 0/2:33,0,35:0.515,0 ("0.515,0" being the AF), while the 4.1.7 example shows 0/2:33,0,35:0.515,. ("0.515,." being the AF)

This substitution of '0' for '.' is what is causing the issue.

Thank you for your help.

**Comment by ldgauthier** [comment]
Sorry, I was looking at allele frequency in the INFO field, not allele fraction in the FORMAT field.  Are these Mutect2 GVCFs?

**Comment by pbrennan13** [comment]
Nope. These are from HaplotypeCaller.

**Comment by ldgauthier** [comment]
Can you share the HaplotypeCaller commandline?

**Comment by pbrennan13** [comment]
Just to clarify, the issue described was not in the gVCF produced by HaplotypeCaller. The issue appears to be introduced during the GenotypeGVCF step.

Below is our HaplotypeCaller Command:

gatk HaplotypeCaller -I {input_bam} -O {output_gvcf} -R {ref} \
--pair-hmm-implementation AVX_LOGLESS_CACHING --emit-ref-confidence GVCF --max-reads-per-alignment-start 0 \
--dbsnp {dbsnp} -L {region}  \
-A AlleleFraction -A AS_BaseQualityRankSumTest -A AS_FisherStrand -A AS_InbreedingCoeff  \
-A AS_MappingQualityRankSumTest -A AS_QualByDepth -A AS_ReadPosRankSumTest -A AS_RMSMap\pingQuality  \
-A AS_StrandOddsRatio -A BaseQuality -A BaseQualityRankSumTest -A ChromosomeCounts -A ClippingRankSumTest \
-A CountNs -A Coverage -A DepthPerAlleleBySample -A DepthPerSampleHC -A ExcessHet -A FisherStrand  \
-A FragmentLength -A GenotypeSummaries -A InbreedingCoeff -A LikelihoodRankSumTest  \
-A MappingQuality -A MappingQualityRankSumTest -A MappingQualityZero -A OrientationBiasReadCounts  \
-A OriginalAlignment -A PossibleDeNovo -A QualByDepth -A ReadPosition -A ReadPosRankSumTest  \
-A ReferenceBases -A RMSMappingQuality -A SampleList -A StrandBiasBySample -A StrandOddsRatio  \
-A TandemRepeat -A UniqueAltReadCount

**Comment by ldgauthier** [comment]
It's weird that the AF originally came out as `0` not `0.0` or anything with more precision.  I bet it's a parsing issue because there's no decimal.  I'm pretty sure I know where to look.  Although htsjdk will pretty-print a zero double as `0.00`, so I'm leaning toward making a change in the AlleleFraction annotation code as well.

**Comment by ldgauthier** [comment]
@pbrennan13 can you get me this same site from proband_mother_duo.HC.g.vcf.gz ?  Were the samples combined using CombineGVCFs?

**Comment by pbrennan13** [comment]
Yep! I used CombingGVCFs.

Here is the command I used:
gatk CombineGVCFs --variant proband.HC.g.vcf.gz --variant mother.HC.g.vcf.gz -R hs38DH.fa --dbsnp dbsnp151_common.hg38.vcf.gz -O proband_mother_duo.HC.g.vcf.gz

Here is the site:
chr4    83598622        .       AT      ATT,ATTT,A,<NON_REF>    .       .       BaseQRankSum=1.26;ClippingRankSum=0.074;DP=145;ExcessHet=3.01;GQ_MEAN=625.00;LikelihoodRankSum=1.34;MQ0=0;MQRankSum=0.00;NCC=0;NCount=0;RAW_MQandDP=522000,145;ReadPosRankSum=1.56   GT:AD:AF:DP:F1R2:F2R1:GQ:PL:SB  ./.:33,0,0,35,0:0.515,0.00:68:12,15,0:21,18,0:99:713,812,1542,812,1542,1542,0,731,731,625,812,1542,1542,731,1542:19,14,14,21 ./.:32,33,2,0,0:0.493,0.030,0.00:67:9,23,0,0:19,8,2,0:99:640,0,588,669,725,1662,728,747,1571,1569,728,747,1571,1569,1569:20,12,21,14

Thank you!

**Comment by genomicsguy** [comment]
Just wondering if this issue has been resolved? Is it an issue in how the commands are being run?

**Comment by droazen** [comment]
@ldgauthier Any update on this one?
